server/apps/publish/archive_publish.py

```diff
@@ -282,48 +302,92 @@ def publish(self, doc, updates, target_output_channels=None):
                 updates['source'] = desk['source'] if desk and desk.get('source', '') \
                     else DEFAULT_SOURCE_VALUE_FOR_MANUAL_ARTICLES
 
+            updates['pubstatus'] = 'canceled' if self.publish_type == 'killed' else 'usable'
```

Comment:
Enums are introduced in Python 3.4. We test against Python 3.3 also and the build will fail with enums. Thats the reason I didn't add enums. Otherwise, I prefer not only for this there are other properties like content state, article types,... where we can use Enums